### PR TITLE
Get AIIDA_CONFIG_FOLDER from AIIDA_PATH environment variable

### PR DESCRIPTION
**Pull request by greschd**
Proposal to address #404. 

Currently doesn't allow selecting the ``.aiida`` folder from ``setup``. It only finds *existing* ``.aiida`` folders -- so if you wanted to install in a different place, you'd have to manually create the ``.aiida`` folder in that place first (and set the ``AIIDA_PATH`` environment variable).

### CODE REVIEW

**Comment by greschd** [comment]
Let me know if you like the proposal, then I'll write some documentation as well.

File: aiida/utils/find_folder.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import os
+import itertools
+try:
+    from pathlib import Path
+except ImportError:
+    from pathlib2 import Path
+
+def find_path(root, dir_name):
+    path = Path(os.path.abspath(root))
+    for p in itertools.chain([path], path.parents):
```
Comment (giovannipizzi) [inline aiida/utils/find_folder.py]: Is this stopping at Filesystem boundaries?

Comment (greschd) [inline aiida/utils/find_folder.py]: currently it doesn't stop at the FS boundary

**Comment by greschd** [comment]
I added the print of the configuration folder to ``verdi profile list``.

**Comment by greschd** [comment]
So are there any opinions on whether we should keep the "going up" in searching for the ``.aiida`` folder?

**Comment by DropD** [comment]
@greschd: Are you still working on this? If you are blocked by the open questions I would suggest leaving away the recursive search and keeping it as simple as possible. Upwards search can always be reintroduced in another PR later on.

It speeds up reviews and approval of PRs in general if they introduce only one feature at a time.

**Comment by greschd** [comment]
If the current behavior (upward search included) is ok, then it is ready to merge. 

After using it for a while on my own, I can say there are certainly times where I would miss not having upward search. Especially the case of having different test and production config works well that way. The question is if there are some concerns about it or not. From my side I don't see any.

**Comment by DropD** [comment]
I don't see any concerns, however I don't feel like i can take that decision alone, since I was not around when the discussion was going on.

I would suggest you resolve the conflicts that exist now and remove the "WIP" from the title in that case.
@giovannipizzi: Will you have time to revisit this, once it is ready?

**Comment by greschd** [comment]
@giovannipizzi The only thing that still needs to be done is adding a description in the documentation. I guess this should go in the "optional configuration" section, correct?

With the current approach to getting the path stopping at FS boundaries is not easily possible, since ``pathlib`` just handles the path as a string.

**Comment by giovannipizzi** [comment]
Yes, that section of the docs is ok!

**Comment by giovannipizzi** [comment]
Thanks!
I'm just wondering if computing the AIIDA_CONFIG_FOLDER at module load time isn't too slow, but if we find that it is the case we can implement some sort of caching.

**Comment by greschd** [comment]
I did do some timing in #404. The conclusion was that it's pretty much negligible. Maybe this changes if you have NFS or something else that's slowing it down significantly, but I would guess in that case the other operations become slower as well.

**Pull request merged**